zebra-state/src/service/tests.rs

```diff
@@ -183,3 +187,87 @@ fn state_behaves_when_blocks_are_committed_out_of_order() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn legacy_chain() -> Result<()> {
+    zebra_test::init();
+
+    legacy_chain_for_network(Network::Mainnet)?;
+    legacy_chain_for_network(Network::Testnet)?;
+
+    Ok(())
+}
```

Comment:
Optional styling: feel free to ignore this, and I apologize for the amount of code in the comment :sweat: 

I thought it might be useful to mention that I try to create separate `#[test]`s for separate checks, so that when it fails it's a bit easier to track down. In this case it might be harder because of testing both testnet and mainnet, but one idea might be to make that part of the proptest input. That also allows gathering the `ProptestConfig::with_cases` configuration into a single place.

I've included an example below (which also includes some changes suggested in other comments below). Just an idea, but it might be useful as inspiration for any other changes :sweat_smile: :

```
const DEFAULT_PARTIAL_CHAIN_PROPTEST_CASES: u32 = 2;
const BLOCKS_AFTER_NU5: u32 = 100;

proptest! {
    #![proptest_config(
        proptest::test_runner::Config::with_cases(env::var("PROPTEST_CASES")
            .ok()
            .and_then(|v| v.parse().ok())
            .unwrap_or(DEFAULT_PARTIAL_CHAIN_PROPTEST_CASES))
    )]

    #[test]
    fn at_least_one_transaction_with_network_upgrade(
        (network, height, chain) in partial_nu5_chain_strategy(5, true, BLOCKS_AFTER_NU5)
    ) {
        let response = crate::service::legacy_chain_check(height, chain.into_iter(), network)
            .map_err(|error| error.to_string());

        prop_assert_eq!(response, Ok(()));
    }

    #[test]
    fn no_transactions_with_network_upgrade(
        (network, height, chain) in partial_nu5_chain_strategy(4, true, BLOCKS_AFTER_NU5 + 1)
    ) {
        let response = crate::service::legacy_chain_check(height, chain.into_iter(), network)
            .map_err(|error| error.to_string());

        prop_assert_eq!(
            response,
            Err("giving up after checking too many blocks".into())
        );
    }

    #[test]
    fn at_least_one_transactions_with_inconsistent_network_upgrade(
        (network, height, chain) in partial_nu5_chain_strategy(5, true, BLOCKS_AFTER_NU5)
    ) {
        let start_check_height = (height - 1).expect("Too few blocks after NU5 activation");
        let response =
            crate::service::legacy_chain_check(start_check_height, chain.into_iter(), network)
                .map_err(|error| error.to_string());

        prop_assert_eq!(
            response,
            Err("inconsistent network upgrade found in transaction".into())
        );
    }
}
```

Reply:
Yea, i think this is a good idea. I applied in https://github.com/ZcashFoundation/zebra/pull/2366/commits/b46c0a6f2408865d9cac034270bdaa4a1221e72f

Had to change some stuff but not because your suggestion was wrong but because the third tests had problems.

With this in place many of the other suggestions do not apply, i am going to mark them as resolved. 

Thanks.